textdata_preprocessor.py

```diff
@@ -0,0 +1,253 @@
+# Copyright 2021 The Petuum Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=attribute-defined-outside-init
+"""
+IU Xray medical report preprocessor for the text that does extraction
+of findings and impression part of the text. Tokenize the sentence, lowercase
+all the characters and remove the word contains non-alphabetic characters
+"""
+from typing import Iterator, Any
+import os.path as osp
+import argparse
+import xml.etree.ElementTree as ET
+from collections import Counter
+
+from ft.onto.base_ontology import Token, Document
+from forte.data.data_pack import DataPack
+from forte.processors.base import PackProcessor
+from forte.data.readers.base_reader import PackReader
+from forte.pipeline import Pipeline
+from forte.processors.nltk_processors import NLTKWordTokenizer
+from forte.data.multi_pack import MultiPack
+from forte.processors.base import MultiPackProcessor
+from forte.processors.writers import PackNameJsonPackWriter
+from forte.data.caster import MultiPackBoxer
+from forte.data.selector import NameMatchSelector
+from forte.data.data_utils_io import dataset_path_iterator
+
+from iu_xray.onto import Impression, Findings, FilePath
+
+
+class FindingsExtractor(PackProcessor):
+    r"""A processor to extract the Findings session in the medical report.
+    """
+
+    def _process(self, input_pack: DataPack):
+        findings_ind = input_pack.text.find("FINDINGS")
```

Comment:
I find out that the source files are different from mimic. In this case, the `FINDINGS` and `IMPRESSION` are structured as XML files. 
https://github.com/petuum/medical_images/pull/8/files#diff-fa0ef15e3d5bfdcc5ae8d747a380b0e5b5dfe92ab9b3d631dae09de29a8d7416R49

So you don't need to use the `find` text method to get the begin and end for them. It is your choice whether to put the `FINDINGS` text in the text content or not, but you should be in control of the text creation, so you should know exactly where the spans of `FINDING` are. So I would suggest you directly add these `FINDING` extraction directly into your reader.

Reply:
That's true! It is natural to directly add these ``FINDINGS`` during reading the XML file

Reply:
Modified